File: language-testing/hodor/hodor.rb
#! /usr/bin/env ruby
# This application replaces all words with hodor.
# Usage: ./hodor.rb 'hello world!'
# Usage: echo "hello world!" | ./hodor.rb

def hodor(str)
  str.gsub(/[a-zA-Z0-9]+/, 'hodor')
end

if ARGV.length > 0
  puts hodor(ARGV.join(' '))
else
  $stdin.each { |line| puts hodor(line) }
end

File: language-testing/hodor/hodor.py
#! /usr/bin/env python3
# This application replaces all words with hodor.
# Usage: ./hodor.py 'hello world!'
# Usage: echo "hello world!" | ./hodor.py

from re import sub
from sys import argv, stdin

def hodor(line):
    print(sub("\w+", "hodor", line), end="\n" if line[-1] != '\n' else "")

def main():
    if len(argv) > 1:
        hodor(" ".join(argv[1:]))
    else:
        for line in stdin:
            hodor(line)

if __name__ == "__main__":
    main()
